tools/html_transformer/html_transformer.js

```diff
@@ -0,0 +1,105 @@
+var Promise = require('es6-promise').Promise;
+var jsdom = require('jsdom');
+
+exports.load = function(filename) {
+  return new Promise(function(resolve, reject) {
+    jsdom.env(filename, function(errors, window) {
+      if (errors && errors.length) {
+        return reject(errors[0]);
+      }
+
+      return resolve(new HTMLTransformer(window));
+    });
+  });
+};
+
+function HTMLTransformer(window) {
```

Comment:
I would like to add some comments about what this component is for, and what it would do at the beginning of a "class".